ui/components/src/UnitSelector/UnitSelector.tsx

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import { Box, FormControl, FormControlLabel, Switch, TextField, Autocomplete, SwitchProps } from '@mui/material';
+import {
+  UnitOptions,
+  UNIT_CONFIG,
+  DEFAULT_DECIMAL_PLACES,
+  UnitConfig,
+  isUnitWithDecimalPlaces,
+  isUnitWithAbbreviate,
+} from '../model';
+
+export interface UnitSelectorProps {
+  value: UnitOptions;
+  onChange: (unit: UnitOptions) => void;
+}
+
+type AutocompleteKindOption = UnitConfig & { id: UnitOptions['kind'] };
+
+const KIND_OPTIONS: AutocompleteKindOption[] = Object.entries(UNIT_CONFIG).map(([id, config]) => {
+  return {
+    id: id as UnitOptions['kind'],
+    ...config,
+  };
+});
+
+const DECIMAL_OPTIONS = [0, 1, 2, 3, 4];
+
+export function UnitSelector({ value, onChange }: UnitSelectorProps) {
+  const hasDecimalPlaces = isUnitWithDecimalPlaces(value);
+  const hasAbbreviate = isUnitWithAbbreviate(value);
+
+  const handleKindChange = (_: unknown, newValue: AutocompleteKindOption) => {
+    onChange({
+      kind: newValue.id,
+    });
+  };
+
+  const handleDecimalChange = (_: unknown, newValue: number) => {
+    if (hasDecimalPlaces) {
+      onChange({
+        ...value,
+        decimal_places: newValue,
+      });
+    }
+  };
+
+  const handleAbbreviateChange: SwitchProps['onChange'] = (_: unknown, checked: boolean) => {
+    if (hasAbbreviate) {
+      onChange({
+        ...value,
+        abbreviate: checked,
+      });
+    }
+  };
+
+  const kindConfig = UNIT_CONFIG[value.kind];
+
+  return (
+    <>
+      <FormControlLabel
+        label="Abbreviate"
+        control={
+          <Switch
+            checked={hasAbbreviate ? value.abbreviate : undefined}
+            onChange={handleAbbreviateChange}
+            disabled={!hasAbbreviate}
+          />
+        }
+      />
+      <FormControl sx={{ minWidth: 200 }}>
+        <Autocomplete
+          value={{ id: value.kind, ...kindConfig }}
+          options={KIND_OPTIONS}
+          isOptionEqualToValue={(option, value) => option.id === value.id}
+          groupBy={(option) => option.group || 'Other'}
+          renderInput={(params) => <TextField {...params} label="Units" />}
+          renderOption={(renderOptsProps, option) => {
+            // Custom option needed to get some increased left padding to make
+            // the items more distinct from the group label.
```

Comment:
I couldn't put the padding directly on the `li` because of some CSS rules deeper in the MUI code that override it.